addons/account/models/account_invoice.py

```diff
@@ -1045,6 +1045,7 @@ def refund(self, date_invoice=None, date=None, description=None, journal_id=None
                 'in_invoice': ('vendor bill refund')}
             message = _("This %s has been created from: <a href=# data-oe-model=account.invoice data-oe-id=%d>%s</a>") % (invoice_type[invoice.type], invoice.id, invoice.number)
             refund_invoice.message_post(body=message)
+            invoice.refund_invoice_id = refund_invoice
```

Comment:
You're right, a migration script would be needed to add the inverse relation on refunded invoices.

But there's another problem here actually. It's currently possible to create multiple refunds for 1 invoice. So using this many2one won't work.

You need to create a new One2many field and use that one.